lang/java/reef-checkpoint/src/main/java/org/apache/reef/io/checkpoint/fs/FSCheckpointID.java

```diff
@@ -34,8 +34,7 @@
 
   private Path path;
 
-  public FSCheckpointID() {
```

Comment:
Ah. Java Serialization requires a `public` default constructor. Even if it is empty:

> The subtype may assume this responsibility only if the class it extends has an accessible no-arg constructor to initialize the class's state. It is an error to declare a class Serializable if this is not the case. The error will be detected at runtime.

[Source](https://docs.oracle.com/javase/8/docs/api/java/io/Serializable.html)